index.js

```diff
@@ -119,9 +114,14 @@ const parseArgs = (
         );
         return result;
       } else if (StringPrototypeCharAt(arg, 1) !== '-') {
-        // Look for shortcodes: -fXzy
+        // Look for shortcodes: -fXzy and expand them to -f -X -z -y:
         if (arg.length > 2) {
-          throw new ERR_NOT_IMPLEMENTED('short option groups');
+          for (let i = 2; i < arg.length; i++) {
+            const short = StringPrototypeCharAt(arg, i);
+            // Add 'i' to 'pos' such that short options are parsed in order
+            // of definition:
+            ArrayPrototypeSplice(argv, pos + (i - 1), 0, `-${short}`);
+          }
```

Comment:
I find this iteration/splice combo a bit confusing. Wondering if we could simplify this by processing `argv` before entering the main while loop.

For example, I've put together an `expandShortcodes` util that doesn't require much thinking.
```js
const expandShortcodes = argv => argv.map(arg => (
  arg?.[0] === '-' && arg?.[1] !== '-'
    ? arg.slice(1).split('').map(short => `-${short}`)
    : arg
)).flat()

const parseArgs = (
  argv = getMainArgs(),
) => {
  argv = expandShortcodes(argv)
  while () {...}
}
```

This is by no means a show stopper, but curious what folks think!

Reply:
![image](https://user-images.githubusercontent.com/32409546/152662582-fc9bf894-feb2-4374-b3d8-2bb9cd536e43.png)

Reply:
@aaronccasanova seems like a reasonable suggestion, although:

```
const expandShortcodes = argv => argv.map(arg => (
  arg?.[0] === '-' && arg?.[1] !== '-'
    ? arg.slice(1).split('').map(short => `-${short}`)
    : arg
)).flat()
```

Would need to be refactored to use primordials, i.e., `ArrayPrototypeMap`, `StringPrototypeSlice`, `StringPrototypeSplit` -- I would be tempted to omit the functional helpers, and just preprocess in a loop.

If other folks like this refactor (@shadowspawn, @ljharb?). Feel like submitting a patch against this PR? I can rebase when I land.

Reply:
I think it is safer to do the work at the point the argument is understood. For example, options after a `--` are treated as positional and should not be expanded. 

(For interest, Commander used to expand the arguments before processing just like you suggest. To support more advanced features correctly, I moved the expansion to inside the loop after other logic had been applied.)

Reply:
Perhaps we could just make the splice logic a little more straight forward, the `-2` is a bit `(i - 1)` is a bit magic, could rename to `currentArgPosition`, or something.

Reply:
> I think it is safer to do the work at the point the argument is understood. For example, options after a -- are treated as positional and should not be expanded.

Great point!

Eh, perhaps we shouldn't dwell on this. The current implementation is great, just requires a bit of thinking 🙂

Reply:
i didn't comment originally, but any code that uses "splice" is always going to be less clear than code that doesn't :-)